## Pandas - Analyzing DataFrames

### Viewing the Data


In [28]:
import pandas as pd

df = pd.read_csv('files/baseball.csv')

print(df.head(10))

# .head() dan .tail() method by default will return the top 5 rows

              Name Team        Position  Height  Weight    Age PosCategory
0    Adam_Donachie  BAL         Catcher      74     180  22.99     Catcher
1        Paul_Bako  BAL         Catcher      74     215  34.69     Catcher
2  Ramon_Hernandez  BAL         Catcher      72     210  30.78     Catcher
3     Kevin_Millar  BAL   First_Baseman      72     210  35.43   Infielder
4      Chris_Gomez  BAL   First_Baseman      73     188  35.71   Infielder
5    Brian_Roberts  BAL  Second_Baseman      69     176  29.39   Infielder
6    Miguel_Tejada  BAL       Shortstop      69     209  30.77   Infielder
7      Melvin_Mora  BAL   Third_Baseman      71     200  35.07   Infielder
8      Aubrey_Huff  BAL   Third_Baseman      76     231  30.19   Infielder
9       Adam_Stern  BAL      Outfielder      71     180  27.05  Outfielder


In [29]:
print(df.tail())

                Name Team        Position  Height  Weight    Age PosCategory
1010   Brad_Thompson  STL  Relief_Pitcher      73     190  25.08     Pitcher
1011   Tyler_Johnson  STL  Relief_Pitcher      74     180  25.73     Pitcher
1012  Chris_Narveson  STL  Relief_Pitcher      75     205  25.19     Pitcher
1013   Randy_Keisler  STL  Relief_Pitcher      75     190  31.01     Pitcher
1014     Josh_Kinney  STL  Relief_Pitcher      73     195  27.92     Pitcher


### Info About the Data

The DataFrames object has a method called `info()`, that gives you more information about the data set.

In [30]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1015 entries, 0 to 1014
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Name         1015 non-null   object 
 1   Team         1015 non-null   object 
 2   Position     1015 non-null   object 
 3   Height       1015 non-null   int64  
 4   Weight       1015 non-null   int64  
 5   Age          1015 non-null   float64
 6   PosCategory  1015 non-null   object 
dtypes: float64(1), int64(2), object(4)
memory usage: 55.6+ KB
None


Result Explained :

The result tells us there are 1015 rows and 7 columns:

```bash
RangeIndex: 1015 entries, 0 to 1014
Data columns (total 7 columns):
```

And the name of each column, with the data type:

```bash
#   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Name         1015 non-null   object 
 1   Team         1015 non-null   object 
 2   Position     1015 non-null   object 
 3   Height       1015 non-null   int64  
 4   Weight       1015 non-null   int64  
 5   Age          1015 non-null   float64
 6   PosCategory  1015 non-null   object
```